clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -0,0 +1,269 @@
+          DescribeTasksResult describeTasksResult = amazonECS.describeTasks(new DescribeTasksRequest().withCluster(ecsClusterName).withTasks(listTasksResult.getTaskArns()));
+
+          for (Task task: describeTasksResult.getTasks()) {
+            InstanceStatus ec2InstanceStatus = containerInformationService.getEC2InstanceStatus(amazonEC2, containerInformationService.getContainerInstance(amazonECS, task));
+            instances.add(new EcsTask(extractTaskIdFromTaskArn(task.getTaskArn()), task, ec2InstanceStatus));
+          }
+        }
+
+        EcsServerGroup ecsServerGroup = generateServerGroup(awsRegion, metadata, instances);
+        EcsServerCluster spinnakerCluster = generateSpinnakerServerCluster(credentials, metadata, loadBalancers, ecsServerGroup);
+
+        if (clusterMap.get(metadata.applicationName) != null) {
+          clusterMap.get(metadata.applicationName).add(spinnakerCluster);
+        } else {
+          clusterMap.put(metadata.applicationName, Sets.newHashSet(spinnakerCluster));
+        }
+      }
+    }
+
+    return clusterMap;
+  }
+
+  private Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> extractLoadBalancersData(DescribeLoadBalancersResult loadBalancersResult) {
+    Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> loadBalancers = Sets.newHashSet();
+    for (LoadBalancer elb: loadBalancersResult.getLoadBalancers()) {
+      AmazonLoadBalancer loadBalancer = new AmazonLoadBalancer();
+      loadBalancer.setName(elb.getLoadBalancerName());
+      loadBalancers.add(loadBalancer);
+    }
+    return loadBalancers;
+  }
+
+  private EcsServerCluster generateSpinnakerServerCluster(AmazonCredentials credentials, ServiceMetadata metadata, Set<com.netflix.spinnaker.clouddriver.model.LoadBalancer> loadBalancers, EcsServerGroup ecsServerGroup) {
+    return new EcsServerCluster()
+          .withAccountName(credentials.getName())
+          .withName(metadata.cloudStack)
+          .withLoadBalancers(loadBalancers)
+          .withServerGroups(Sets.newHashSet(ecsServerGroup));
+  }
+
+  private EcsServerGroup generateServerGroup(AmazonCredentials.AWSRegion awsRegion, ServiceMetadata metadata, Set<Instance> instances) {
+    return new EcsServerGroup()
+          .withName(constructServerGroupName(metadata))
+          .withCloudProvider("aws")   // TODO - Implement ECS in Deck so we can stop tricking the front-end app here
+          .withType("aws")            // TODO - Implement ECS in Deck so we can stop tricking the front-end app here
+          .withRegion(awsRegion.getName())
+          .withInstances(instances);
+  }
+
+  private String constructServerGroupName(ServiceMetadata metadata) {
+    return metadata.applicationName + "-" +  metadata.cloudStack + "-" + metadata.serverGroupVersion; // TODO - support CLOUD_DETAIL variable
+  }
+
+  private ServiceMetadata extractMetadataFromServiceArn(String arn) {
```

Comment:
Use a regex to capture the different groups and raise error on lack of matched
https://docs.oracle.com/javase/tutorial/essential/regex/

Extra points for documenting what you expect the Arn to look like next to the regex string.

Reply:
This will not be carried over in the refactoring into a Caching Agent